framework/src/actions/AdaptivityAction.C

```diff
@@ -170,15 +170,15 @@ AdaptivityAction::act()
 
     adapt.setPrintMeshChanged(getParam<bool>("print_changed_info"));
 
-    const std::vector<std::string> & weight_names =
-        getParam<std::vector<std::string>>("weight_names");
-    const std::vector<Real> & weight_values = getParam<std::vector<Real>>("weight_values");
+    if (isParamValid("weight_names") && isParamValid("weight_values"))
```

Comment:
can you use a default instead? for both params

Reply:
Here are test failed with default value.
`mesh/adapt.displaced_test ................................................................... FAILED (EXODIFF)
executioners/executioner.test_steady_adapt .................................................. FAILED (EXODIFF)
outputs/displaced.use_displaced ............................................................. FAILED (EXODIFF)
time_integrators/crank-nicolson.group/adapt ................................................. FAILED (EXODIFF)
mesh/adapt.test_time ........................................................................ FAILED (EXODIFF)
preconditioners/pbp.pbp/pbp_adapt_test ...................................................... FAILED (EXODIFF)
time_integrators/implicit-euler.group/adapt ................................................. FAILED (EXODIFF)
preconditioners/smp.smp/smp_adapt_test ...................................................... FAILED (EXODIFF)
time_integrators/dirk.group/dirk-2d-heat-adap ............................................... FAILED (EXODIFF)
time_integrators/bdf2.group/adapt ........................................................... FAILED (EXODIFF)
mesh/adapt.initial_adaptivity_test .......................................................... FAILED (EXODIFF)
preconditioners/reuse.mesh_refinement ....................................................... FAILED (CSVDIFF)
mesh/adapt.patch_test ....................................................................... FAILED (EXODIFF)
mesh/adapt.adapt_cycles_test ................................................................ FAILED (EXODIFF)
mesh/adapt.test ............................................................................. FAILED (EXODIFF)
bcs/periodic.testlevel1 ..................................................................... FAILED (EXODIFF)
mesh/adapt.interval ......................................................................... FAILED (EXODIFF)`

Reply:
but why? they had a default value before.

Reply:
> but why? they had a default value before. 

I think they may have 'isParamValid' check downstream. If we use default values, some of them may need change to ParamSetByUser or size check. I don't dig in because there will be a lot of tests and more(if you consider all the apps). For this PR, I just want to make sure this new changes won't break all the existing cases and code. I try not to change their logic if possible

Reply:
They dont, I checked. 
I just did this with the {} default (e.g. reverted to what it was):
```
    // if (isParamValid("weight_names") && isParamValid("weight_values"))
    // {
    const std::vector<std::string> & weight_names =
        getParam<std::vector<std::string>>("weight_names");
    const std::vector<Real> & weight_values = getParam<std::vector<Real>>("weight_values");

    auto num_weight_names = weight_names.size();
    auto num_weight_values = weight_values.size();

    if (num_weight_names != num_weight_values)
      mooseError("Number of weight_names must be equal to number of weight_values in "
                 "Execution/Adaptivity");

    if (num_weight_names)
    {
```
and it worked fine, 0 test failures, with a default

Reply:
I see, the main difference here is use default and check size vs use empty value and check isParamValid. I think both logics work under this condition. Why it has to be size check instead of isParamValid() check?

Reply:
Because passing {}, an empty vector that has true for isParamValid (and if the user specifies "" for the parameter, it also gets true for isParamSetByUser), is not valid for the code below the if